boa_tester/src/main.rs

```diff
@@ -785,11 +785,9 @@ struct TestResult {
     #[serde(rename = "v", default)]
     edition: SpecEdition,
     #[serde(rename = "s", default)]
-    strict: bool,
-    #[serde(skip)]
-    result_text: Box<str>,
-    #[serde(rename = "r")]
-    result: TestOutcomeResult,
+    strict: Option<TestOutcomeResult>,
+    #[serde(rename = "r", default)]
+    no_strict: Option<TestOutcomeResult>,
```

Comment:
We could simplify this by making `TestOutcomeResult` something like:
```Rust
#[serde(tag = "t", content = "d")
enum TestOutcomeResult {
    Passed,
    Ignored,
    Failed {
        strict: bool,
        reason: String
    },
    Panic {
        strict: bool,
        reason: String,
    }
}
```
Then, we won't need separate fields for `strict` and `no_strict`, just a single `result`.

Reply:
I was thinking about that, and also about the file size & JS performance in the browser. I think that having dictionaries with test/suite names as keys would decrease the size of the JSON around a 15%.

Booleans occupy a bit more. Maybe adding more variants, such as "StrictFailed" and so on could be more performant.

Reply:
Maybe it's time to transition to a binary serialization format such as [MessagePack](https://msgpack.org/index.html)?

Another idea I had was to use [Postcard](https://crates.io/crates/postcard) and a small wasm glue code to deserialize the postcard and serialize it to JSON, then we would call that function from the conformance page.

Reply:
I'll play a bit with those

Reply:
I've given this a thought, and it could be a nice enhancement in the future, but I would for now just use JSON until the new website is up and running. What do you think?

Reply:
Yep, we can revisit this when we finish the site redesign.

Putting that aside, I still think it would be better to have `FailedStrict` and `PanicStrict` for simplicity.